sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1594,6 +1594,15 @@ object SQLConf {
         "WHERE, which does not follow SQL standard.")
       .booleanConf
       .createWithDefault(false)
+
+  val LEGACY_ALIAS_NON_STRUCT_GROUPING_KEY =
+    buildConf("spark.sql.legacy.dataset.aliasNonStructGroupingKey")
```

Comment:
Maybe aliasNonStructGroupingKeyAsValue, and default to true.

Then we can remove this in the future.

Reply:
Ok. That makes sense. Thanks.